xarray/plot/utils.py

```diff
@@ -265,7 +265,13 @@ def _determine_cmap_params(plot_data, vmin=None, vmax=None, cmap=None,
     if extend is None:
         extend = _determine_extend(calc_data, vmin, vmax)
 
-    if levels is not None or isinstance(norm, mpl.colors.BoundaryNorm):
+    cmap_is_colormap = (
+        isinstance(cmap, mpl.colors.LinearSegmentedColormap)
+        or isinstance(cmap, mpl.colors.ListedColormap)
+    )
```

Comment:
Both ihnerit from `mpl.colors.Colormap`, and after a quick check it seems these are the only two in mpl, so it might be possible to test for this only

Reply:
OK. done. This is what I tried first but it didn't work; perhaps there was some other bug. The example works now and the tests pass locally.